resources/aws/gateway.go

```diff
@@ -103,15 +88,18 @@ func (g *Gateway) CreateOrFail() error {
 }
 
 func (g *Gateway) Delete() error {
-	gateway, err := g.findExisting()
+	gateways, err := g.list()
 	if err != nil {
 		return microerror.MaskAny(err)
 	}
 
-	if _, err := g.Clients.EC2.DeleteInternetGateway(&ec2.DeleteInternetGatewayInput{
-		InternetGatewayId: gateway.InternetGatewayId,
-	}); err != nil {
-		return microerror.MaskAny(err)
+	for _, gateway := range gateways {
+		_, err := g.Clients.EC2.DeleteInternetGateway(&ec2.DeleteInternetGatewayInput{
+			InternetGatewayId: gateway.InternetGatewayId,
+		})
+		if err != nil {
+			return microerror.MaskAny(err)
+		}
 	}
```

Comment:
But IMO the more important problem which I care about more is dealing with situations when the list is empty. Here, in the `Delete` method. I think that we should return an error to inform user about that in logs. That was my intention when I intoduced the "coudnt' find..." errors - to just log that. To simplify debugging.